### PR TITLE
replace direct go-libp2p-core references with go-libp2p/core

### LINKED ISSUE #762: Use libp2p/go-libp2p/core instead of legacy libp2p/go-libp2p-core
We should use the new format of the package as the old one has been archived

### PR DESCRIPTION
**Pull request by neelvirdy**
Fixes https://github.com/filecoin-project/go-fil-markets/issues/762

### CODE REVIEW

**Comment by neelvirdy** [comment]
@hannahhoward @dirkmc could i get a review please?

File: go.mod [hunk 1]
```diff
@@ -106,6 +105,7 @@ require (
 	github.com/libp2p/go-buffer-pool v0.1.0 // indirect
 	github.com/libp2p/go-cidranger v1.1.0 // indirect
 	github.com/libp2p/go-libp2p-asn-util v0.2.0 // indirect
+	github.com/libp2p/go-libp2p-core v0.20.1 // indirect
```
Comment (marten-seemann) [inline go.mod:108]: At some point, this will also need to disappear. This is coming from a transitive dependency, so it's fine for now.

**Review by marten-seemann — APPROVED** [review]

**Comment by neelvirdy** [comment]
@marten-seemann i dont have perms to merge, mind granting me if you can or merging this?

**Comment by neelvirdy** [comment]
@hannahhoward @marten-seemann @dirkmc pinging again to get help with merging since i dont have perms

**Comment by dirkmc** [comment]
Making changes to go-fil-markets means we also need to make the same change in boost and lotus, then run a comprehensive test suite on both of those projects, so it's not a simple task.
Is there an urgent need to make this change?

**Comment by marten-seemann** [comment]
> Making changes to go-fil-markets means we also need to make the same change in boost and lotus, then run a comprehensive test suite on both of those projects, so it's not a simple task. Is there an urgent need to make this change?

This PR doesn't bump any versions, so it shouldn't have any side effects on other repos. Or am I missing something?

**Comment by marten-seemann** [comment]
@dirkmc @hannahhoward Is there anything else needed to get this PR merged?

**Comment by dirkmc** [comment]
@marten-seemann is the code at
```
"github.com/libp2p/go-libp2p-core/peer"
```
exactly the same as the code at
```
"github.com/libp2p/go-libp2p/core/peer"
```
?

Same question for protocol, network, etc

**Comment by marten-seemann** [comment]
Yes, we [moved](https://github.com/libp2p/go-libp2p/pull/1683) go-libp2p-core into go-libp2p a while back.

**Pull request merged**